packages/@o3r/components/builders/metadata-check/index.it.spec.ts

```diff
@@ -0,0 +1,378 @@
+  packageJson = {
+    ...packageJson,
+    name: packageName,
+    private: false
+  };
+  await writeFileAsJSON(packageJsonPath, packageJson);
+
+  // Set old metadata and publish to registry
+  await writeFileAsJSON(metadataPath, previousConfigurationMetadata);
+
+  let latestVersion;
+  try {
+    latestVersion = getLatestPackageVersion(packageName, execAppOptionsWorkspace);
+  } catch {
+    latestVersion = baseVersion;
+  }
+
+  const bumpedVersion = inc(latestVersion, 'patch');
+
+  const args = getPackageManager() === 'yarn' ? [] : ['--no-git-tag-version', '-f'];
+  packageManagerVersion(bumpedVersion, args, execAppOptions);
+
+  packageManagerPublish([], execAppOptions);
+
+  // Override with new metadata for comparison
+  await writeFileAsJSON(metadataPath, newMetadata);
+
+  // Add migration data file
+  await writeFileAsJSON(migrationDataPath, migrationData);
+};
+
+describe('check metadata migration', () => {
+  beforeEach(async () => {
+    const { applicationPath } = o3rEnvironment.testEnvironment;
+    const execAppOptions = { ...getDefaultExecSyncOptions(), cwd: applicationPath };
+    await promises.copyFile(
+      join(__dirname, '..', '..', '..', '..', '..', '.verdaccio', 'conf', '.npmrc'),
+      join(applicationPath, '.npmrc')
+    );
+    execFileSync('npx', [
+      '--yes',
+      'npm-cli-login',
+      '-u',
+      'verdaccio',
+      '-p',
+      'verdaccio',
+      '-e',
+      'test@test.com',
+      '-r',
+      'http://127.0.0.1:4873'
+    ], execAppOptions);
+  });
+
+  test('should not throw', async () => {
+    await initTest(
+      true,
+      newConfigurationMetadata,
+      defaultMigrationData,
+      'allow-breaking-changes'
+    );
+    const { workspacePath, appName } = o3rEnvironment.testEnvironment;
+    const execAppOptionsWorkspace = { ...getDefaultExecSyncOptions(), cwd: workspacePath };
+
+    expect(() => packageManagerExec({ script: 'ng', args: ['run', `${appName}:check-metadata`] }, execAppOptionsWorkspace)).not.toThrow();
+  });
+
+  test('should throw because no migration data', async () => {
+    await initTest(
+      true,
+      newConfigurationMetadata,
+      {
+        ...defaultMigrationData,
+        changes: []
+      },
+      'no-migration-data'
+    );
+    const { workspacePath, appName } = o3rEnvironment.testEnvironment;
+    const execAppOptionsWorkspace = { ...getDefaultExecSyncOptions(), cwd: workspacePath };
+
+    try {
+      packageManagerExec({ script: 'ng', args: ['run', `${appName}:check-metadata`] }, execAppOptionsWorkspace);
+      throw new Error('should have thrown before');
+    } catch (e: any) {
+      expect(e.message).not.toBe('should have thrown before');
+      previousConfigurationMetadata.slice(1).forEach(({ library, name, properties }) => {
+        expect(e.message).toContain(`Property ${library}#${name} ${properties[0]!.name} has been modified but is not documented in the migration document`);
+      });
+    }
```

Comment:
I really think that doing try/catch in a test is bad idea
Would something like this works ?
```suggestion
    const expectedErrors = new RegExp(previousConfigurationMetadata.slice(1).map(({ library, name, properties }) =>
      escapeRegExp(`Property ${library}#${name} ${properties[0]!.name} has been modified but is not documented in the migration document`)
    ).join('.*'), 'm');
    expect(() => 
      packageManagerExec({ script: 'ng', args: ['run', `${appName}:check-metadata`] }, execAppOptionsWorkspace);
   ).toThrow(expectedErrors);
 
```

Reply:
as discussed it more complex to check several message and even more complex with the non presence of a message